site/docs/master/configuration.md

```diff
@@ -1,9 +1,54 @@
 # Contour Configuration
 
+## Configuration File
+
 A configuration file can be passed to the `contour serve` command which specified additional properties that Contour should use when starting up.
 This file is passed to Contour via a ConfigMap which is mounted as a volume to the Contour pod.
 
-Following is an example ConfigMap with configuration file included: 
+The Contour configuration file is optional.
+In its absence, Contour will operate with reasonable defaults.
+
+| Field Name | Type | Description |
+|------------|------|-------------|
+| accesslog-format | string | This key sets the global [access log format][2] for Envoy. Valid options are `envoy` or `json`. |
+| debug | boolean | Enables debug logging. |
+| disablePermitInsecure | boolean | If this field is true, Contour will ignore `PermitInsecure` field in HTTPProxy documents. |
+| incluster | boolean | This field specifies that Contour is running in a Kubernetes cluster and should use the in-cluster client access configuration.  |
+| json-fields | string array | This is list the field names to include in the JSON [access log format][2]. |
+| kubeconfig | string | Path to a Kubernetes [kubeconfig file][3]. This should not be needed in the vast majority of deployments. |
```

Comment:
nit: Maybe change the vast majority of deployments line to say what it's used for? Like, only used when running outside of the cluster?